python-sgd-regression/sgd_regression.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+
+# QUESTION: when creating files in python-sgd-regression I copied folder python-linear-regression and just renamed
+# `linear-regression` to `sgd-regression`. Is there another way? e.g. cookiecutter template or something?
+
+# QUESTION: I'd like to create some python unit tests in py.test (to speed up development), any suggestions where should
+# I put them?
+
+from io_helper import io_helper
+from db_models import JobResult
+
+import logging
+import json
+
+import pandas as pd
+from sklearn.linear_model import SGDRegressor
+import patsy
+from sqlalchemy import create_engine
+from sqlalchemy.orm import sessionmaker
+
+
+DEFAULT_DOCKER_IMAGE = "python-sgd-regression"
+
+
+def main():
+    # Configure logging
+    logging.basicConfig(level=logging.INFO)
+
+    job_id = io_helper._get_job_id()
+    partial_job_id = None
+
+    # Start reading stream of inputs
+    # QUESTION: the tables `cde_*` are just for a proof of concept? How is the data going to be distributed in
+    # production? Probably not as tables in a single database (in that case we wouldn't need all the serialization
+    # of models and storing intermediate results in `job_result` table). In separate databases? Or is this just a
+    # preparation for federated learning where each docker container (running on hospital's servers)
+    # will save its intermediate results? Thanks for explanation!
+
+    # QUESTION: related to the question above - if we indeed have all the tables in a single DB, why do we have to
+    # store intermediate results to `job_result` table? Can't we just keep everything in memory?
+
```

Comment:
Intermediate results are going to be copied from one hospital to another, and Woken will store the results gathered from a previous hospital into the job_results table of the local hospital before calling the algorithm.